### PR TITLE
[Numpy] allow mix integer dtypes for power/add/multiply

### PR DESCRIPTION
**Pull request by JiangZhaoh**
## Description ##
resolution for issue [#17839](https://github.com/apache/incubator-mxnet/issues/17839)
allow to mix bool\int8\int32\int64 for add\multiply\power
Examples:
````
import mxnet as mx
from mxnet import numpy as np
mx.npx.set_np()
a = np.ones((10,), dtype=mx.np.int32)
b = np.ones((10,), dtype=mx.np.int64)
c = a * b
c
c = a + b
c
c = np.power(a, b)
c
````
Print:
```
array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1], dtype=int64)
array([2, 2, 2, 2, 2, 2, 2, 2, 2, 2], dtype=int64)
array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1], dtype=int64)

```
### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/operator/numpy/np_elemwise_broadcast_op.h [hunk 1]
```diff
@@ -51,6 +51,11 @@ inline void PrintErrorMessage(const std::string& op_name, const int dtype1, cons
              << " yet...";
 }
 
+inline bool is_integer(const int dtype) {
```
Comment (haojin2) [inline src/operator/numpy/np_elemwise_broadcast_op.h]: no need for this extra function, you could use `common::is_float(const int dtype)` instead.

**Comment by JiangZhaoh** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by JiangZhaoh** [comment]
@mxnet-bot run ci [unix-gpu]
